Cannot build on Ubuntu 16.04

**Describe the bug**
Following instructions from docs

**To Reproduce**
Steps to reproduce the behavior:
1. Clone the Synfig repo as stated in [docs](https://synfig-docs-dev.readthedocs.io/en/latest/building/Preparing%20Environment.html)
2. Execute ./1-setup-linux-native.sh
3. Execute ./2-build-debug.sh
4. Gives error

```
In file included from /usr/include/sigc++-2.0/sigc++/signal_base.h:27:0,
                 from /usr/include/sigc++-2.0/sigc++/signal.h:8,
                 from /usr/include/sigc++-2.0/sigc++/sigc++.h:86,
                 from /usr/include/glibmm-2.4/glibmm/signalproxy.h:12,
                 from /usr/include/glibmm-2.4/glibmm/objectbase.h:23,
                 from /usr/include/glibmm-2.4/glibmm/wrap.h:23,
                 from /usr/include/glibmm-2.4/glibmm/containerhandle_shared.h:26,
                 from /usr/include/glibmm-2.4/glibmm/arrayhandle.h:23,
                 from /usr/include/glibmm-2.4/glibmm/miscutils.h:25,
                 from /home/veer/synfig.git/ETL/ETL/_stringf.h:45,
                 from /home/veer/synfig.git/ETL/ETL/stringf:34,
                 from /home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:41:
/usr/include/sigc++-2.0/sigc++/functors/slot.h: In instantiation of ‘class sigc::slot0<bool(synfig::Node*)>’:
/usr/include/sigc++-2.0/sigc++/functors/slot.h:1655:7:   required from ‘class sigc::slot<bool(synfig::Node*)>’
/home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:1544:39:   required from here
/usr/include/sigc++-2.0/sigc++/functors/slot.h:431:42: error: function returning a function
   typedef T_return (*call_type)(rep_type*);
                                          ^
/usr/include/sigc++-2.0/sigc++/functors/slot.h:437:19: error: function returning a function
   inline T_return operator()() const
                   ^
/usr/include/sigc++-2.0/sigc++/functors/slot.h: In instantiation of ‘class sigc::slot0<bool(const synfig::Node*)>’:
/usr/include/sigc++-2.0/sigc++/functors/slot.h:1655:7:   required from ‘class sigc::slot<bool(const synfig::Node*)>’
/home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:1544:39:   required from here
/usr/include/sigc++-2.0/sigc++/functors/slot.h:431:42: error: function returning a function
   typedef T_return (*call_type)(rep_type*);
                                          ^
/usr/include/sigc++-2.0/sigc++/functors/slot.h:437:19: error: function returning a function
   inline T_return operator()() const
                   ^
In file included from /usr/include/sigc++-2.0/sigc++/signal_base.h:27:0,
                 from /usr/include/sigc++-2.0/sigc++/signal.h:8,
                 from /usr/include/sigc++-2.0/sigc++/sigc++.h:86,
                 from /usr/include/glibmm-2.4/glibmm/signalproxy.h:12,
                 from /usr/include/glibmm-2.4/glibmm/objectbase.h:23,
                 from /usr/include/glibmm-2.4/glibmm/wrap.h:23,
                 from /usr/include/glibmm-2.4/glibmm/containerhandle_shared.h:26,
                 from /usr/include/glibmm-2.4/glibmm/arrayhandle.h:23,
                 from /usr/include/glibmm-2.4/glibmm/miscutils.h:25,
                 from /home/veer/synfig.git/ETL/ETL/_stringf.h:45,
                 from /home/veer/synfig.git/ETL/ETL/stringf:34,
                 from /home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:41:
/usr/include/sigc++-2.0/sigc++/functors/slot.h: In instantiation of ‘struct sigc::internal::slot_call0<synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>::<lambda(synfig::Node*)>, bool(synfig::Node*)>’:
/usr/include/sigc++-2.0/sigc++/functors/slot.h:454:82:   required from ‘sigc::slot0<T_return>::slot0(const T_functor&) [with T_functor = synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/usr/include/sigc++-2.0/sigc++/functors/slot.h:1668:26:   required from ‘sigc::slot<T_return, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil>::slot(const T_functor&) [with T_functor = synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:1544:39:   required from here
/usr/include/sigc++-2.0/sigc++/functors/slot.h:104:19: error: function returning a function
   static T_return call_it(slot_rep* rep)
                   ^
/usr/include/sigc++-2.0/sigc++/functors/slot.h: In instantiation of ‘struct sigc::internal::slot_call0<synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>, bool(synfig::Node*)>’:
/usr/include/sigc++-2.0/sigc++/functors/slot.h:454:82:   required from ‘sigc::slot0<T_return>::slot0(const T_functor&) [with T_functor = synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/usr/include/sigc++-2.0/sigc++/functors/slot.h:1668:26:   required from ‘sigc::slot<T_return, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil>::slot(const T_functor&) [with T_functor = synfig::Canvas::invoke_signal_value_node_child_removed(etl::handle<synfig::ValueNode>, etl::handle<synfig::ValueNode>)::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:1552:54:   required from here
/usr/include/sigc++-2.0/sigc++/functors/slot.h:104:19: error: function returning a function
/usr/include/sigc++-2.0/sigc++/functors/slot.h: In instantiation of ‘struct sigc::internal::slot_call0<synfig::Node::find_all_parents_of_type() const [with T = synfig::Layer_PasteCanvas]::<lambda(synfig::Node*)>, bool(synfig::Node*)>’:
/usr/include/sigc++-2.0/sigc++/functors/slot.h:454:82:   required from ‘sigc::slot0<T_return>::slot0(const T_functor&) [with T_functor = synfig::Node::find_all_parents_of_type() const [with T = synfig::Layer_PasteCanvas]::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/usr/include/sigc++-2.0/sigc++/functors/slot.h:1668:26:   required from ‘sigc::slot<T_return, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil, sigc::nil>::slot(const T_functor&) [with T_functor = synfig::Node::find_all_parents_of_type() const [with T = synfig::Layer_PasteCanvas]::<lambda(synfig::Node*)>; T_return = bool(synfig::Node*)]’
/home/veer/synfig.git/synfig-core/src/synfig/node.h:282:3:   required from ‘std::vector<etl::handle<T> > synfig::Node::find_all_parents_of_type() const [with T = synfig::Layer_PasteCanvas]’
/home/veer/synfig.git/synfig-core/src/synfig/canvas.cpp:121:103:   required from here
/usr/include/sigc++-2.0/sigc++/functors/slot.h:104:19: error: function returning a function
  CXX      libsynfig_la-loadcanvas.lo
  CXX      libsynfig_la-main.lo
Makefile:2697: recipe for target 'libsynfig_la-canvas.lo' failed
make[3]: *** [libsynfig_la-canvas.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:53:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:224:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<_xmlSAXHandler> sax_handler_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:54:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/parsers/textreader.h:260:10: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
     std::auto_ptr<PropertyReader> propertyreader;
          ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:69:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/relaxngschema.h:108:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<Impl> pimpl_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:70:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/xsdschema.h:99:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<Impl> pimpl_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:73:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/validators/relaxngvalidator.h:159:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<Impl> pimpl_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
In file included from /usr/include/libxml++-2.6/libxml++/libxml++.h:75:0,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/libxml++-2.6/libxml++/validators/xsdvalidator.h:149:8: warning: ‘template<class> class std::auto_ptr’ is deprecated [-Wdeprecated-declarations]
   std::auto_ptr<Impl> pimpl_;
        ^
In file included from /usr/include/c++/5/memory:81:0,
                 from /usr/include/libxml++-2.6/libxml++/parsers/saxparser.h:14,
                 from /usr/include/libxml++-2.6/libxml++/libxml++.h:53,
                 from /home/veer/synfig.git/synfig-core/src/synfig/loadcanvas.cpp:49:
/usr/include/c++/5/bits/unique_ptr.h:49:28: note: declared here
   template<typename> class auto_ptr;
                            ^
make[3]: Leaving directory '/home/veer/synfig.git/_debug/synfig-core/src/synfig'
Makefile:465: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/veer/synfig.git/_debug/synfig-core/src'
Makefile:687: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/veer/synfig.git/_debug/synfig-core'
Makefile:555: recipe for target 'all' failed
make: *** [all] Error 2
```

**System information:**
 - OS: Ubuntu
 - OS version: 16.04
 - Commit ID: 80aabd4

--- a/synfig-core/src/synfig/node.h
+++ b/synfig-core/src/synfig/node.h
@@ -264,8 +264,8 @@ public:
 
 	//! Callback function for a foreach method.
 	//! If it returns true, the foreach iteration is halted.
-	using ForeachFunc = sigc::slot<bool(Node*)>;
-	using ConstForeachFunc = sigc::slot<bool(const Node*)>;
+	using ForeachFunc = sigc::slot1<bool, Node*>;
+	using ConstForeachFunc = sigc::slot1<bool, const Node*>;
 
 	//! Call function func for each of the parents of the current Node
 	//! Do not add or remove any parent node while doing this foreach call
@@ -302,7 +302,7 @@ public:
 	//! Return the first parent of a given type with an additional match condition
 	//! The CompareFunc should return true when match is satisfied
 	//! Example: node->find_first_parent_of_type<MyType>([](MyType::Handle v) -> bool { return v.prop == 1;});
-	template<typename T> etl::handle<T> find_first_parent_of_type(const sigc::slot<bool(const etl::handle<T>&)> &CompareFunc) const {
+	template<typename T> etl::handle<T> find_first_parent_of_type(sigc::slot1<bool, const etl::handle<T>&> CompareFunc) const {
 		T* parent = nullptr;
 		foreach_parent([&parent, CompareFunc](const Node* node) -> bool {
 			if (auto item = dynamic_cast<T*>(const_cast<Node*>(node)))

--- a/synfig-core/src/synfig/soundprocessor.cpp
+++ b/synfig-core/src/synfig/soundprocessor.cpp
@@ -292,7 +292,8 @@ bool SoundProcessor::subsys_init() {
 bool SoundProcessor::subsys_stop()
 {
 #ifndef WITHOUT_MLT
-	delete Internal::repository;
+	// delete leads to crash on Ubuntu 16.04
+	//delete Internal::repository;
 	Mlt::Factory::close();
 #endif
 	return true;

--- a/synfig-studio/src/gui/dials/jackdial.h
+++ b/synfig-studio/src/gui/dials/jackdial.h
@@ -55,7 +55,7 @@ class JackDial : public Gtk::Box
 public:
 	JackDial();
 	void set_state(bool enabled);
-	Glib::SignalProxy<void> signal_toggle_jack() { return toggle_jack_button->signal_toggled(); }
+	Glib::SignalProxy0<void> signal_toggle_jack() { return toggle_jack_button->signal_toggled(); }
 
 	sigc::signal<void>& signal_offset_changed()      { return offset_widget->signal_value_changed(); }
 

--- a/synfig-studio/src/gui/dials/keyframedial.h
+++ b/synfig-studio/src/gui/dials/keyframedial.h
@@ -55,8 +55,8 @@ public:
 
 	KeyFrameDial();
 	void on_mode_changed(synfigapp::EditMode mode); // Updates button icons/state
-	Glib::SignalProxy<void> signal_toggle_keyframe_past() { return toggle_keyframe_past->signal_toggled(); }
-	Glib::SignalProxy<void> signal_toggle_keyframe_future() { return toggle_keyframe_future->signal_toggled(); }
+	Glib::SignalProxy0<void> signal_toggle_keyframe_past() { return toggle_keyframe_past->signal_toggled(); }
+	Glib::SignalProxy0<void> signal_toggle_keyframe_future() { return toggle_keyframe_future->signal_toggled(); }
 
 }; // END of class KeyFrameDial
 

